### PR TITLE
Implement PATCH

### PR DESCRIPTION
**Pull request by Cryptex-github**
- [x] Get message history
- [x] Edit channel
- [x] Edit guild
- [x] Edit message
- [x] Edit user

### CODE REVIEW

**Review by hydrostaticcog — COMMENTED** [review]
General syntax looks good, though I have not tested this. I differ to @tazz4843 to evaluate the code quality.

**Review by hydrostaticcog — DISMISSED** [review]
Build Failed

**Comment by Cryptex-github** [comment]
> Build Failed

That most likely caused by sqlx for not having the latest `sql-data.json`

**Comment by tazz4843** [comment]
Thanks for the PR. As we speak, I'm testing out this on a migrated DB.

File: ferrischat_webserver/src/channels/edit_channel.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    let bigint_channel_id = u128_to_bigdecimal!(channel_id);
+
+    let db = get_db_or_fail!();
+
+    let ChannelUpdateJson { name } = channel_info.0;
+
+    let resp = sqlx::query!(
+        "UPDATE channels SET name = $2 WHERE id = $1 RETURNING *",
+        bigint_channel_id,
+        name
```
Comment (tazz4843) [inline ferrischat_webserver/src/channels/edit_channel.rs:22]: This ordering feels weird. Swap `$1` and `$2` in both the query and macro, please.

File: ferrischat_webserver/src/guilds/edit_guild.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    let bigint_guild_id = u128_to_bigdecimal!(guild_id);
+
+    let GuildUpdateJson { name } = guild_info.0;
+
+    let db = get_db_or_fail!();
+
+    let resp = sqlx::query!(
+        "UPDATE guilds SET name = $2 WHERE id = $1 RETURNING *",
+        bigint_guild_id,
+        name
```
Comment (tazz4843) [inline ferrischat_webserver/src/guilds/edit_guild.rs:22]: Same as above, please swap `$1` and `$2` in both the query and macro.

File: ferrischat_webserver/src/messages/edit_message.rs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            Err(err) => {
+                return HttpResponse::InternalServerError().json(InternalServerErrorJson {
+                    reason: format!("DB returned an error: {}", err),
+                })
+            }
+        }
+    }
+
+    let MessageUpdateJson { content } = message_info.0;
+    let resp = sqlx::query!("UPDATE messages SET content = $3, edited_at = now()::timestamp without time zone WHERE channel_id = $1 AND id = $2 RETURNING *", bigint_message_id, bigint_channel_id, content)
```
Comment (tazz4843) [inline ferrischat_webserver/src/messages/edit_message.rs:51]: Change query order to be $1, then $2, and then $3, please.

**Review by tazz4843 — CHANGES_REQUESTED** [review]
Query order is pretty much my only issue with this.

File: ferrischat_webserver/src/messages/edit_message.rs (hunk 3 above)
Comment (Cryptex-github) [reply ferrischat_webserver/src/messages/edit_message.rs:51]: fine

File: ferrischat_webserver/src/guilds/edit_guild.rs (hunk 2 above)
Comment (Cryptex-github) [reply ferrischat_webserver/src/guilds/edit_guild.rs:22]: fine

File: ferrischat_webserver/src/channels/edit_channel.rs (hunk 1 above)
Comment (Cryptex-github) [reply ferrischat_webserver/src/channels/edit_channel.rs:22]: fine

**Review by tazz4843 — APPROVED** [review]

**Pull request merged**